translib/transformer/xfmr_intf.go

```diff
@@ -0,0 +1,2143 @@
+		return result, errors.New("Invalid interface type IntfTypeUnset")
+	}
+	if IntfTypeEthernet != intfType {
+		return result, nil
+	}
+	intTbl := IntfTypeTblMap[intfType]
+
+	tblName, _ := getPortTableNameByDBId(intTbl, inParams.curDb)
+	pTbl := data[tblName]
+	prtInst := pTbl[inParams.key]
+	autoNeg, ok := prtInst.Field[PORT_AUTONEG]
+	if ok {
+		if autoNeg == "on" || autoNeg == "true" {
+			result["auto-negotiate"] = true
+		} else {
+			result["auto-negotiate"] = false
+		}
+	} else {
+		log.Info("auto-negotiate field not found in DB")
+	}
+	return result, err
+}
+
+var DbToYang_intf_eth_port_speed_xfmr FieldXfmrDbtoYang = func(inParams XfmrParams) (map[string]interface{}, error) {
+	var err error
+	result := make(map[string]interface{})
+
+	data := (*inParams.dbDataMap)[inParams.curDb]
+	intfType, _, ierr := getIntfTypeByName(inParams.key)
+	if intfType == IntfTypeUnset || ierr != nil {
+		log.Info("DbToYang_intf_eth_port_speed_xfmr - Invalid interface type IntfTypeUnset")
+		return result, errors.New("Invalid interface type IntfTypeUnset")
+	}
+
+	intTbl := IntfTypeTblMap[intfType]
+
+	tblName, _ := getPortTableNameByDBId(intTbl, inParams.curDb)
+	pTbl := data[tblName]
+	prtInst := pTbl[inParams.key]
+	speed, ok := prtInst.Field[PORT_SPEED]
+	portSpeed := ocbinds.OpenconfigIfEthernet_ETHERNET_SPEED_UNSET
+	if ok {
+		portSpeed, err = getDbToYangSpeed(speed)
+		result["port-speed"] = ocbinds.E_OpenconfigIfEthernet_ETHERNET_SPEED.ΛMap(portSpeed)["E_OpenconfigIfEthernet_ETHERNET_SPEED"][int64(portSpeed)].Name
+	} else {
+		log.Info("Speed field not found in DB")
+	}
+
+	return result, err
+}
+
+var intf_post_xfmr PostXfmrFunc = func(inParams XfmrParams) (map[string]map[string]db.Value, error) {
+
+	requestUriPath := (NewPathInfo(inParams.requestUri)).YangPath
+	retDbDataMap := (*inParams.dbDataMap)[inParams.curDb]
+	log.Info("Entering intf_post_xfmr")
+	log.Info(requestUriPath)
+	xpath, _, _ := XfmrRemoveXPATHPredicates(inParams.requestUri)
+
+	if inParams.oper == DELETE {
+
+		err_str := "Delete not allowed at this container"
+		/* Preventing delete at IPv6 config level*/
+		if xpath == "/openconfig-interfaces:interfaces/interface/subinterfaces/subinterface/ipv6/config" {
+			log.Info("In interface Post transformer for DELETE op ==> URI : ", inParams.requestUri)
+			return retDbDataMap, tlerr.NotSupported(err_str)
+		}
+
+		/* For delete request and for fields with default value, transformer adds subOp map with update operation (to update with default value).
+		   So, adding code to clear the update SubOp map for delete operation to go through for the following requestUriPath */
+		if xpath == "/openconfig-interfaces:interfaces/interface/subinterfaces/subinterface/ipv6/config/enabled" {
+			if len(inParams.subOpDataMap) > 0 {
+				dbMap := make(map[string]map[string]db.Value)
+				if inParams.subOpDataMap[4] != nil && (*inParams.subOpDataMap[4])[db.ConfigDB] != nil {
```

Comment:
we should use const for brevity
http://10.59.132.240:8023/sonic_4.0.x/xref/sonic-mgmt-common/translib/transformer/xconst.go#73
const (
	GET Operation = iota + 1
	CREATE
	REPLACE
	UPDATE
	DELETE
	SUBSCRIBE
	MAXOPER
)